### PR TITLE
Refactor hard code in rest_test.go

### PR DESCRIPTION
**Pull request by hanxiaoshuai**
**What this PR does / why we need it**:
Refactor hard code in rest_test.go
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest_test.go [hunk 1]
```diff
@@ -922,3 +818,67 @@ func TestFinishRequest(t *testing.T) {
 		}
 	}
 }
+
+func (tc *patchTestCase) setTcStartingPod(name string, namespace string, uid types.UID, resourceVersion string, apiVersion string, activeDeadlineSeconds *int64, nodeName string) {
+	tc.startingPod.Name = name
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest_test.go]: all 4 funcs are the same, with a different pod. Please factor the common code out.

Comment (hanxiaoshuai) [reply staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest_test.go]: Done, Thanks for your review.

**Pull request merged**